tests/test_client.py

```diff
@@ -1496,3 +1497,66 @@ def test_dropped_transaction(sentry_init, capture_record_lost_event_calls, test_
 def test_enable_tracing_deprecated(sentry_init, enable_tracing):
     with pytest.warns(DeprecationWarning):
         sentry_init(enable_tracing=enable_tracing)
+
+
+def make_options_transport_cls():
+    """Make an options transport class that captures the options passed to it."""
+    # We need a unique class for each test so that the options are not
+    # shared between tests.
+
+    class OptionsTransport(Transport):
+        """Transport that captures the options passed to it."""
+
+        def __init__(self, options):
+            super().__init__(options)
+            type(self).options = options
+
+        def capture_envelope(self, _):
+            pass
+
+    return OptionsTransport
+
+
+@contextlib.contextmanager
+def clear_env_var(name):
+    """Helper to clear the a given environment variable,
+    and restore it to its original value on exit."""
+    old_value = os.environ.pop(name, None)
+
+    try:
+        yield
+    finally:
+        if old_value is not None:
+            os.environ[name] = old_value
+        elif name in os.environ:
+            del os.environ[name]
+
+
+@pytest.mark.parametrize(
+    ("env_value", "arg_value", "expected_value"),
+    [
+        (None, None, False),  # default
+        ("0", None, False),  # env var false
+        ("1", None, True),  # env var true
+        (None, False, False),  # arg false
+        (None, True, True),  # arg true
+        # Argument overrides environment variable
+        ("0", True, True),  # env false, arg true
+        ("1", False, False),  # env true, arg false
```

Comment:
Do you all think this order of precedence is reasonable? Or, should we not guarantee any specific order of precedence (i.e. if both are set to different values, behavior is undefined)?

Reply:
I suppose the middle option would be to remove the tests asserting order and not provide any specific guarantee on precedence for now.

That way we could change the precedence later even in a minor release

Reply:
But we have an order right now: Options set when calling `init()` are overruling env vars.

And I think we can not change this in a minor release, because it can alter behavior for users that have a init option and an env var set.

the tests are good, because if we change the precedence by accident, the tests will fail.

Reply:
sounds good, then we stay with what we currently have